charmhelpers/core/decorators.py

```diff
@@ -53,3 +53,34 @@ def _retry_on_exception_inner_2(*args, **kwargs):
         return _retry_on_exception_inner_2
 
     return _retry_on_exception_inner_1
+
+
+def retry_on_predicate(num_retries, predicate_fun, base_delay=0):
+    """Retry based on return value
+
+    The return value of the decorated function is passed to the given predicate_fun. If the
+    result of the predicate is False, retry the decorated function up to num_retries times
+
+    An exponential backoff up to base_delay^num_retries seconds can be introduced by setting
+    base_delay to a nonzero value
+    """
```

Comment:
Need do describe the parameters:

:param num_retries: ..
:type num_retries: int

etc.